sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/TooManyGenericParameters.cs

```diff
@@ -90,23 +90,19 @@ protected override void Initialize(ParameterLoadingAnalysisContext context)
                 SyntaxKind.MethodDeclaration);
         }
 
-        private static string GetTypeKeyword(TypeDeclarationSyntax typeDeclaration)
-        {
-            switch (typeDeclaration.Kind())
+        private static string GetTypeKeyword(TypeDeclarationSyntax typeDeclaration) =>
+            typeDeclaration.Kind() switch
             {
-                case SyntaxKind.ClassDeclaration:
-                    return "class";
-
-                case SyntaxKind.StructDeclaration:
-                    return "struct";
-
-                case SyntaxKind.InterfaceDeclaration:
-                    return "interface";
+                SyntaxKind.ClassDeclaration => "class",
+                SyntaxKind.StructDeclaration => "struct",
+                SyntaxKind.InterfaceDeclaration => "interface",
+                _ => GetUnknownType(typeDeclaration)
```

Comment:
The coverage is low due to this line. The default branch is not reachable since we explicitly subscribe only to class, struct and interface declaration.

I would not remove this though since one could subscribe also for records and forget to update this switch.